tracing-appender/src/non_blocking.rs

```diff
@@ -0,0 +1,102 @@
+use crate::worker::Worker;
+use crossbeam_channel::{bounded, Sender};
+use std::borrow::Borrow;
+use std::io;
+use std::sync::atomic::AtomicU64;
+use std::sync::atomic::Ordering;
+use std::sync::Arc;
+use std::thread::JoinHandle;
+use tracing_subscriber::fmt::MakeWriter;
+
+
+pub struct NonBlocking {
+    writer: NonBlockingWriter,
+    worker_guard: JoinHandle<()>,
+    error_counter: Arc<AtomicU64>,
+}
+
+#[derive(Clone, Debug)]
+pub struct NonBlockingWriter {
+    channel: Sender<Vec<u8>>,
+    error_counter: Arc<AtomicU64>,
+}
+
+impl NonBlocking {
+    pub fn new<'a, T: MakeWriter + Send + Sync + 'static>(make_writer: T) -> NonBlocking {
+        NonBlockingBuilder::default().build(make_writer)
+    }
+
+    fn create<'a, T: MakeWriter + Send + Sync + 'static>(
+        sender: Sender<Vec<u8>>,
+        error_counter: Arc<AtomicU64>,
+        worker: Worker<T>,
+    ) -> NonBlocking {
+        Self {
+            writer: NonBlockingWriter {
+                channel: sender,
+                error_counter: error_counter.clone(),
+            },
+            worker_guard: worker.worker_thread(),
+            error_counter
+        }
+    }
+
+    pub fn writer(&self) -> NonBlockingWriter {
+        self.writer.clone()
+    }
+
+    pub fn error_counter(&self) -> Arc<AtomicU64> {
+        self.error_counter.clone()
+    }
+}
+
+pub struct NonBlockingBuilder {
+    buffered_lines_limit: usize,
+}
+
+impl NonBlockingBuilder {
+    pub fn buffered_lines_limit(mut self, buffered_lines_limit: usize) -> NonBlockingBuilder {
```

Comment:
I believe this is emitting an unused method warning because the builder type is not publicly exported. It's `pub`, but the module it's in is private. We should probably make the `non_blocking` module public, so that we have a place for the builder to live.